assets/src/design-system/components/tooltip/index.js

```diff
@@ -0,0 +1,233 @@
+import styled, { css } from 'styled-components';
+import { useState, useRef, useMemo, useEffect } from 'react';
+/**
+ * Internal dependencies
+ */
+import { THEME_CONSTANTS } from '../../theme';
+import { Text } from '../typography';
+import { TOOLTIP_POSITIONS } from './constants';
+
+const SVG_TOOLTIP_TAIL_ID = 'tooltip-tail';
+const TAIL_WIDTH = 34;
+const TAIL_HEIGHT = 8;
+const SPACE_BETWEEN_TOOLTIP_AND_ELEMENT = 8;
+
+const SvgForTail = styled.svg`
+  position: absolute;
+  width: 0;
+  height: 0;
+`;
+
+const getTailPosition = (position) => {
+  switch (position) {
+    case TOOLTIP_POSITIONS.TOP_CENTER:
+      return css`
+        bottom: -${TAIL_HEIGHT}px;
+        left: calc(50% - ${TAIL_WIDTH / 2}px);
+        transform: rotate(180deg);
+      `;
+
+    case TOOLTIP_POSITIONS.TOP_LEFT:
+      return css`
+        bottom: -${TAIL_HEIGHT}px;
+        left: 0;
+        transform: rotate(180deg);
+      `;
+
+    case TOOLTIP_POSITIONS.TOP_RIGHT:
+      return css`
+        bottom: -${TAIL_HEIGHT}px;
+        right: 0;
+        transform: rotate(180deg);
+      `;
+
+    case TOOLTIP_POSITIONS.BOTTOM_RIGHT:
+      return css`
+        top: -${TAIL_HEIGHT}px;
+        right: 0;
+      `;
+
+    case TOOLTIP_POSITIONS.BOTTOM_LEFT:
+      return css`
+        top: -${TAIL_HEIGHT}px;
+        left: 0;
+      `;
+
+    // default is TOOLTIP_POSITIONS.BOTTOM_CENTER
+    default: {
+      return css`
+        top: -${TAIL_HEIGHT}px;
+        left: calc(50% - ${TAIL_WIDTH / 2}px);
+      `;
+    }
+  }
+};
+const Tail = styled.span`
+  @supports (clip-path: url('#${SVG_TOOLTIP_TAIL_ID}')) {
+    position: absolute;
+    display: block;
+    height: ${TAIL_HEIGHT}px;
+    width: ${TAIL_WIDTH}px;
+    ${({ position }) => getTailPosition(position)};
+    background-color: inherit;
+    border: inherit;
+    clip-path: url('#${SVG_TOOLTIP_TAIL_ID}');
+  }
+`;
+
+export const Content = styled.div(
+  ({ theme, visible }) => css`
+    visibility: ${visible ? 'visible' : 'hidden'};
+    position: absolute;
+    max-width: 300px;
+    border-radius: 4px;
+    padding: 10px;
+    background-color: ${theme.colors.interactiveBg.primaryNormal};
+    opacity: ${visible ? 1 : 0};
+
+    transition: opacity linear 200ms;
+  `
+);
+
+const TooltipText = styled(Text)`
+  color: ${({ theme }) => theme.colors.bg.primary};
+  overflow-wrap: break-word;
+`;
+
+const ContentWrapper = styled.div`
+  display: inline-flex;
+`;
+
+const Container = styled.div.attrs({
+  ['data-testid']: 'tooltip-container',
+})`
+  position: relative;
+  display: inline-flex;
+  height: inherit;
+  width: inherit;
+`;
+
+function Tooltip({
+  children,
+  content,
+  hasTail,
+  position = TOOLTIP_POSITIONS.BOTTOM_CENTER,
+}) {
+  const containerRef = useRef();
+  const contentRef = useRef();
+  const previousContent = useRef();
+  const [showTooltip, setShowTooltip] = useState(false);
+
+  useEffect(() => {
+    if (content != previousContent.current) {
+      setShowTooltip(false);
```

Comment:
So this is to dismiss the tooltip if the content changes?

Reply:
yea I'm not following what this is for either. Maybe a good place to add a comment and explain the purpose of it

Reply:
It serves to keep tooltip current. It's only called initially in storybook. Taking it out keeps the same functionality in storybook, but this was added in the dashboard's tooltip and I rather think it has a purpose in keeping the position calculated and the tooltip current that are cases in storybook I can't stress test. I want to leave it in and play with it more when subbing out the tooltips in the dashboard. Then, if it proves still useful, adding a note in so that we can maintain why this is important.

Reply:
makes sense. sounds good!

Reply:
https://github.com/google/web-stories-wp/pull/5331#discussion_r526256595 

it feels like it's not doing anything. @carlos-kelly implemented this first in the dashboard and i figured it's for some edge case that's hard to notice otherwise. Maybe I take it out and see what happens and if anything bad we can implement a solution.

Reply:
@BrittanyIRL @littlemilkstudio 

It was an edge case and the case was:

When you would click this icon it would instantly show the tooltip for the toggle button. We didn't want that to happen. It's an edge case because I can't think of another place where a button with a tooltip unmounts a component and immediately replaces it with another button with a tooltip of different content.

<img width="209" alt="Screen Shot 2020-11-18 at 2 17 30 PM" src="https://user-images.githubusercontent.com/1738349/99583252-d5043780-29a8-11eb-8c6f-88f310ea90e2.png">
hovering and showing the tooltip) 

<img width="244" alt="Screen Shot 2020-11-18 at 2 17 36 PM" src="https://user-images.githubusercontent.com/1738349/99583250-d46ba100-29a8-11eb-948d-2496fcca7c72.png">